### PR TITLE
fix(DB): Importing Copper nodes from VMangos

### LINKED ISSUE #6167: Copper Vein half submerged/in wrong location
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [ ] Alliance
- [x] Horde

##### CONTENT PHASE:

- [ ] Generic
- [x] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

Original report: https://github.com/chromiecraft/chromiecraft/issues/743

##### CURRENT BEHAVIOUR:

Copper Vein GUID 5110 in Durotar (Thunder Ridge area) spawns half submerged and maybe in the wrong location.  It spawns out in the open instead of on cliffs/mounds/rocky areas/etc

WoWScrnShot_052821_163722

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Ore veins don't spawn out in the open like this and are not half submerged

##### STEPS TO REPRODUCE THE PROBLEM:

1.  Go to 945.3, -4123.0 in Durotar (Thunder Ridge)
2.  Copper Vein GUID 5110 spawns in an odd place/half submerged

##### EXTRA NOTES:

NA

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/3716ddf3e4e1ef138df334d01980fea019272001

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

Confirmed and reproduced on a local AC server (AC HASH/COMMIT: ea76481):

WoWScrnShot_060121_050050

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/98815905-copper-vein-half-submerged-in-wrong-location?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #6679: Westfall, Jangolode Mine - Copper Node Clipping Through Minecart
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [x] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

Original report: https://github.com/chromiecraft/chromiecraft/issues/1047

A Copper Vein spawned in the destroyed mine cart near the entrance of the Jangolode Mine in Westfall. Unfortunately I forgot about the command to show the guid for that object. Sorry, next time.

WoWScrnShot_070121_172932

Maybe: [#6647](https://github.com/azerothcore/azerothcore-wotlk/pull/6647) will close this issue.

Tester: Confirmed, GUID is 5378 from entry 1731.

WoWScrnShot_070321_021641

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Nodes shouldn't spawn where they don't belong.

##### STEPS TO REPRODUCE THE PROBLEM:

1. .go o 5378

##### EXTRA NOTES:

N/A.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/cecbd604d3f57312165cc53f8ed40feae51ee45f

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/99646252-westfall-jangolode-mine-copper-node-clipping-through-minecart?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by claudiodfc**
### Improved Copper Vein spawns and pooling for the following zones:
1. Duskwood
2. Westfall
3. Darkshore
4. Loch Modan
5. Wetlands
6. Hillsbrad
7. Silverpine Forest
8. Barrens
9. Thousand Needles
10. Mulgore
11. Durotar
12. Ashenvale
13. Desolace
14. Stonetalon Mountains
15. Redridge Mountains
16. Tirisfal Glades
17. Elwynn Forest
18. Dun Morogh

## Issues Addressed:

- Closes #6167
- Closes #6679

## Tests Performed:

Tested on Windows 10

## How to Test the Changes:

1. Roam around these zones trying to find unusual behaviors on nodes.
2. Make sure The Draenei and Blood elf zones still have their Copper veins

## TODO
- [X] Still need to re-insert the TBC Copper nodes after being deleted

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by claudiodfc** [comment]
@FrancescoBorzi Ready to be tested 👍🏼

File: data/sql/updates/pending_db_world/rev_1624998621825147000.sql [hunk 1]
```diff
@@ -0,0 +1,4105 @@
...
+OR `entry`= @FREE_POOL_TEMPLATE+8
+OR `entry`= @FREE_POOL_TEMPLATE+9
+OR `entry`= @FREE_POOL_TEMPLATE+10
+OR `entry`= @FREE_POOL_TEMPLATE+11
+OR `entry`= @FREE_POOL_TEMPLATE+12
+OR `entry`= @FREE_POOL_TEMPLATE+13
+OR `entry`= @FREE_POOL_TEMPLATE+14
+OR `entry`= @FREE_POOL_TEMPLATE+15
+OR `entry`= @FREE_POOL_TEMPLATE+16
+OR `entry`= @FREE_POOL_TEMPLATE+17;
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1624998621825147000.sql:28]:
```suggestion
DELETE FROM `pool_template` WHERE `entry` BETWEEN @FREE_POOL_TEMPLATE AND @FREE_POOL_TEMPLATE+17;
```

File: data/sql/updates/pending_db_world/rev_1624998621825147000.sql [hunk 2]
```diff
@@ -0,0 +1,4105 @@
...
+(@FREE_POOL_TEMPLATE+15, 15, 'Copper Veins in Tirisfal 45 objects total'),
+(@FREE_POOL_TEMPLATE+16, 28, 'Copper Veins in Elwynn Forest 86 objects total'),
+(@FREE_POOL_TEMPLATE+17, 25, 'Copper Veins in Dun Morogh 76 objects total');
+
+-- Cleaning up gameobjects for inserting. Starting at ID 93000 for all the Copper Veins
+DELETE FROM `gameobject` WHERE `id` =1731
+OR `id`= 3763
+OR `id`= 103713
+OR `id`= 2055
+OR `id`= 181248;
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1624998621825147000.sql:55]:
```suggestion
DELETE FROM `gameobject` WHERE `id` IN (1731, 3763, 103713, 2055, 181248);
```

File: data/sql/updates/pending_db_world/rev_1624998621825147000.sql [hunk 3]
```diff
@@ -0,0 +1,4105 @@
...
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, 'Copper Vein'),
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, 'Copper Vein'),
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, 'Copper Vein'),
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, 'Copper Vein'),
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, 'Copper Vein'),
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, 'Copper Vein'),
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, 'Copper Vein'),
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, 'Copper Vein'),
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, 'Copper Vein'),
+(@FREE_GAMEOBJECT := @FREE_GAMEOBJECT+1, @FREE_POOL_TEMPLATE+10, 0, '
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1624998621825147000.sql:3879]: you can change even these deletes as I did in the previous ones

**Comment by temperrr** [comment]
Basic observation: Any zone where there are more than just copper veins, the other veins are just spawning (see screenshot). Also some places are completely lacking copper which should not happen (see notes). I know this PR focussus on Copper Veins, but in this state its not ready to be merged due to issues with the other ore nodes (that I did not see on AC). 

Test procedure: I ran around the zones, randomly, looking for veins. I tested respawn behaviour in Dun Morogh and Elwynn.

- [ ]     Duskwood - Failed: All veins are spawned at the same time, multiple different veins on top of eachother (none of which are copper) but this did not happen on CC. All over the zone
Screenshot from 2021-07-11 08-57-13

- [ ]     Westfall - Failed ; Only tin veins in Jangolode mine now, should be a healthy mix
- [ ]     Loch Modan - Failed, same thing as in Duskwood, multiple non-copper veins on top of eachother, all over the zone.
- [x]     Wetlands
- [ ]     Hillsbrad - Failed: Same thing as in Duskwood, multiple non-copper veins on top of eachother, all over the zone.
- [x]     Silverpine Forest
- [x]     Barrens
- [x]     Thousand Needles
- [x]     Mulgore
- [x]     Durotar
- [x]     Ashenvale
- [ ]     Desolace - Saw only 7 nodes added - did not find any in my random pattern. Is probably because I didn't look hard enough.
- [x]     Stonetalon Mountains
- [x]     Redridge Mountains
- [x]     Tirisfal Glades
- [x]     Elwynn Forest
- [x]     Dun Morogh

Extra tests:


- [x] Azuremyst
- [x] Bloodmyst
- [x] Eversong Woods
- [x] Ghostlands

**Comment by syssneck** [comment]
Hey, any update here? @temperrr @claudiodfc

**Comment by Nefertumm** [comment]
What's the status for this one? If no response is given, I'll close the PR.

**Comment by claudiodfc** [comment]
Thanks for the test @temperrr 👍🏼 

I have news on this, I'll tag you on another PR.

**Pull request closed without merging**